connmgr/connmanager.go

```diff
@@ -346,6 +352,25 @@ out:
 				log.Debugf("Failed to connect to %v: %v",
 					connReq, msg.err)
 				cm.handleFailedConn(connReq)
+
+			case handleCancelPending:
+				found := false
+				var idToRemove uint64
+				for id, req := range pending {
+					if msg.addr.String() == req.Addr.String() {
+						idToRemove = id
+						found = true
+						break
+					}
+
+				}
+				if found {
+					delete(pending, idToRemove)
```

Comment:
You need to also call `connReq.updateState(connCanceled)` so that if the connection succeeds (and it was previously marked as permanent) the manager won't try to maintain the connection alive.

Reply:
New commit adds the call